mmdet/core/bbox/assigners/atss_cost_assigner.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from ..builder import BBOX_ASSIGNERS
+from ..iou_calculators import build_iou_calculator
+from .assign_result import AssignResult
+from .base_assigner import BaseAssigner
+
+
+@BBOX_ASSIGNERS.register_module()
+class ATSSCostAssigner(BaseAssigner):
+    """Assign a corresponding gt bbox or background to each bbox.
+
+    ATSSCostAssigner is similar with ATSSAssigner, however ATSSCostAssigner
+    use a parameter as a cost rate to distinguish each proposals apparently. 
+
+    Each proposals will be assigned with `0` or a positive integer
+    indicating the ground truth index.
+
+    - 0: negative sample, no assigned gt
+    - positive integer: positive sample, index (1-based) of assigned gt
+
+    Args:
+        topk (float): number of bbox selected in each level
+        alpha (float): param of cost rate for each proposal
+        iou_calculator (dict): builder of IoU calculator
+        ignore_iof_thr (int): whether ignore max overlaps or not(1 or -1)  
+    """
+
+    def __init__(self,
+                 topk,
+                 alpha=0.8,
+                 iou_calculator=dict(type='BboxOverlaps2D'),
+                 ignore_iof_thr=-1):
+        self.topk = topk
+        self.alpha = alpha
+        self.iou_calculator = build_iou_calculator(iou_calculator)
+        self.ignore_iof_thr = ignore_iof_thr
+
```

Comment:
delete a blank line here. Please install `pre-commit` (https://pre-commit.com/), it will automatically fix the format problems like this. If you have difficulty in installing `pre-commit`, leave a comment.

Reply:
I have install `pre-commit` in mmdetection project. Thanks for your suggestion.

> (open-mmlab) pc@pc-Irvingao:~/workspace/torch_ws/ddod_mmdet/mmdetection$ pre-commit install
pre-commit installed at .git/hooks/pre-commit